gravitybee/__init__.py

```diff
@@ -458,7 +458,7 @@ def _create_hook(self):
 
         # 1 - extra data
         hook += "# collection extra data, if any (using --extra-data option)"
-        for data in self.args.extra["data"]:
+        for data in self.args.extra["data"] or []:
```

Comment:
@salman2135 This is solving a *different* problem than the original patch. My understanding is that you have a KeyError on `self.args.extra` because the `data` key is not present. To fix that, use `.get()`:

```suggestion
        for data in self.args.extra.get("data", []):
```

This approach with `or []` is doing something entirely different. That implies that `self.args.extra["data"]` *does* exist, but that it's value is false-y (i.e. `""`, `False`, `{}`, `[]`, or `None`). I don't think this would be a valid config anyway, so this edge case isn't something to solve.

Reply:
Ah I see your point:). Yes, that makes more sense.